tests/proxy_test_helper.c

```diff
@@ -309,7 +309,7 @@ int proxy_tester_send_connect_response(struct proxy_tester *tester) {
     if (tester->failure_type == PTFT_CONNECT_REQUEST) {
         response_string = "HTTP/1.0 401 Unauthorized\r\n\r\n";
     } else {
-        response_string = "HTTP/1.0 200 Connection established\r\n\r\n";
+        response_string = "HTTP/1.0 200 Connection established\r\nconnection: close\r\n";
```

Comment:
Comment that the connection: close header is an edge case

Is it worth having one explicit test where we send this response? instead of sending it in every single test?

Reply:
If these tests were less tangled together in a crazy web of reused functions, I'd say sure. But as they're not, I don't think it's worth the refactor that would entail, and we just want to make sure the behavior is exercised.

Added the comment.